tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -377,7 +372,7 @@ private void ReportWafInitInfoOnce(object sender, InstrumentationGatewaySecurity
         {
             _instrumentationGateway.EndRequest -= ReportWafInitInfoOnce;
             var span = e.RelatedSpan.Context.TraceContext.RootSpan ?? e.RelatedSpan;
-            span.SetTraceSamplingPriority(_settings.KeepTraces ? SamplingPriorityValues.UserKeep : SamplingPriorityValues.AutoReject);
+            span.SetTraceSamplingPriority(SamplingPriorityValues.UserKeep);
```

Comment:
This got changed because `KeepTraces` was removed then added back. I think it's good to keep the first trace with extra information even if `KeepTraces` is set to false, it's it's large volumes of UserKeeps we're worried about, one won't make a difference.